src/Workspaces/Core/Portable/Shared/Collections/IntervalTree`2.cs

```diff
@@ -17,17 +16,22 @@ namespace Microsoft.CodeAnalysis.Shared.Collections
     /// [start, end).  It allows you to efficiently find all intervals that intersect or overlap 
     /// a provided interval.
     /// </summary>
-    internal partial class IntervalTree<T, TIntrospector> : IEnumerable<T>
-        where TIntrospector : struct, IIntervalIntrospector<T>
+    internal partial class IntervalTree<T> : IEnumerable<T>
```

Comment:
basically, the base intervaltree type is specifically not dependent on its introspector (that's what SimpleIntervalTree represents).  Instead, it's just the raw data, without any requirement on the consumer.  the introspector then comes into play in terms of the operations performed on the interval tree.

Reply:
Note: would hae to look, but it may be that thsi approach is no longer needed.  i.e. we don't have people querying with different introspectors than before.